lib/ast.js

```diff
@@ -1418,6 +1418,34 @@ var AST_This = DEFNODE("This", null, {
     },
 }, AST_Symbol);
 
+var AST_Template = DEFNODE("Template", "expressions strings tag", {
```

Comment:
Although I'd prefer mixing strings and expressions in alternating fashion in a single array (strings on even indexes, and expressions on odd), this array separation may have some useful characteristics. I would however suggest to use the order: `tag strings expressions` to better match how they'd be presented. This is a minor point.

Reply:
Just imitating the function signature of tag functions, which will make optimisation easier in the future:
```js
tag`foo${a}bar${b}`
```
⬇️
```js
tag([`foo`, `bar`, ``], a, b)
```
As for the fields, I've been using alphabetical order elsewhere for consistency. May be I can add
```
tag`str1${expr1}...strN${sxprN}strN+1`
```
to `$documentation` for clarification?

Reply:
That doc line addition would be good. I see it's verified in validate.

Reply:
Done in https://github.com/mishoo/UglifyJS/pull/4601/commits/d5f3bcbc361862810ca8b0aabf3a01bb3e8203f9